csrc/fusion.cpp

```diff
@@ -255,7 +255,9 @@ void Fusion::addInput(Val* input) {
 
   if (input->getValType().value() == ValType::TensorView) {
     auto tv = input->as<TensorView>();
-    tv->setMemoryType(MemoryType::Global);
+    if (tv->getMemoryType() != MemoryType::Symmetric) {
+      tv->setMemoryType(MemoryType::Global);
```

Comment:
In reality, the user defines a single-GPU fusion (and calls addInput/addOutput) first and then add things like DeviceMesh, MemoryType and multi-GPU schedules. So this change can be avoided if you are willing to modify your unit tests slightly.

Reply:
ok, np
I will still leave an assertion to avoid wasting time debugging next time.